src/Action.ts

```diff
@@ -79,41 +83,52 @@ export class ActionBase {
-    if (action.actionType === ActionTypes.TEXT) {
-      /**
-       * For backwards compatibility check if payload is of new TextPayload type
-       * Ideally we would implement schema refactor:
-       * 1. Make payloads discriminated unions (E.g. After checking the action.type, flow control knows the type of the payload property)
-       * This removes the need for the GetPayload function and GetArguments which are brittle coding patterns.
-       */
-      try {
-        const textPayload = JSON.parse(action.payload) as TextPayload
-        return EntityIdSerializer.serialize(textPayload.json, entityValues)
-      } catch (e) {
-        const error = e as Error
-        throw new Error(
-          `Error when attempting to parse text action payload. This might be an old action which was saved as a string.  Please create a new action. ${
-          error.message
-          }`
-        )
-      }
-    } else if (action.actionType === ActionTypes.END_SESSION) {
-      const textPayload = JSON.parse(action.payload) as TextPayload
-      return EntityIdSerializer.serialize(textPayload.json, entityValues)
-    }
-    // For API or CARD the payload field of the outer payload is the name of API or the filename of the card template without extension
-    else if (ActionTypes.CARD === action.actionType) {
-      let cardPayload = JSON.parse(action.payload) as CardPayload
-      return cardPayload.payload
-    } else if (ActionTypes.API_LOCAL === action.actionType) {
-      let actionPayload = JSON.parse(action.payload) as ActionPayload
-      return actionPayload.payload
-    } else if (ActionTypes.DISPATCH === action.actionType) {
-      // TODO: Another reason to schema refactor...
-      let actionPayload = JSON.parse(action.payload) as DispatchPayload
-      return `${ActionTypes.DISPATCH}: ${actionPayload.modelName}`
+    switch (action.actionType) {
+      case ActionTypes.TEXT:
+        {
+          /**
+           * For backwards compatibility check if payload is of new TextPayload type
+           * Ideally we would implement schema refactor:
+           * 1. Make payloads discriminated unions (E.g. After checking the action.type, flow control knows the type of the payload property)
+           * This removes the need for the GetPayload function and GetArguments which are brittle coding patterns.
+           */
+          try {
+            const textPayload = JSON.parse(action.payload) as TextPayload
+            return EntityIdSerializer.serialize(textPayload.json, entityValues)
+          } catch (e) {
+            const error = e as Error
+            throw new Error(
+              `Error when attempting to parse text action payload. This might be an old action which was saved as a string.  Please create a new action. ${
+              error.message
+              }`
+            )
+          }
+        }
+      case ActionTypes.END_SESSION:
```

Comment:
Note that the brackets are actually optional for switch statements in Typescript.  Style-wise we've been choosing not to use them

Reply:
IIRC there was some warning about variable redeclaration without them here.  Case statements apparently don't create block scope in JS / TS.

Reply:
ES6 doesn't create separate scopes for each case statements unless you make them different scope by using braces: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch#Block-scope_variables_within_switch_statements

Reply:
Yeah, you can't re-use var names if you don't use brackets.  I usually opt for using diff var names